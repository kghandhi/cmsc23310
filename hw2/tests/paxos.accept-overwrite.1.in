# This example shows a case where an acceptor (Acceptor 1)
# accepts a value, but it is possible for a majority of
# other acceptors to accept a different value.
#
# Basically, Proposer 1 proposes the value 42, which
# is accepted by Acceptor 1. However, before Acceptor 2 and 3
# receive the ACCEPT messages, both Proposer 1 and Acceptor 1
# fail.
# 
# Then, Proposer 2 (presumably detecting that no progress is being made)
# proposes the "null value" zero. However, since Acceptor 1
# has failed, knowledge of the previously accepted knowledge
# has been lost. However, this is fine, because a *majority*
# of acceptors has not yet accepted that value.
#
# The outcome will simply be that the "null value" will be
# agreed upon.

2 3 25
0 PROPOSE 1 42
8 FAIL PROPOSER 1
8 FAIL ACCEPTOR 1
10 PROPOSE 2 0
0 END
