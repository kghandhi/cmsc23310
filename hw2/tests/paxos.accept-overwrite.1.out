  0:    -> P1  PROPOSE v=42
  1: P1 -> A1  PREPARE n=1
  2: P1 -> A2  PREPARE n=1
  3: P1 -> A3  PREPARE n=1
  4: A1 -> P1  PROMISE n=1 (Prior: None)
  5: A2 -> P1  PROMISE n=1 (Prior: None)
  6: A3 -> P1  PROMISE n=1 (Prior: None)
  7: P1 -> A1  ACCEPT n=1 v=42
  8: ** P1 FAILS **
  8: ** A1 FAILS **
  9:
 10:    -> P2  PROPOSE v=0
 11: P2 -> A2  PREPARE n=2
 12: P2 -> A3  PREPARE n=2
 13: A2 -> P2  PROMISE n=2 (Prior: None)
 14: A3 -> P2  PROMISE n=2 (Prior: None)
 15: P2 -> A2  ACCEPT n=2 v=0
 16: P2 -> A3  ACCEPT n=2 v=0
 17: A2 -> P2  ACCEPTED n=2 v=0
 18: A3 -> P2  ACCEPTED n=2 v=0
 19:
 20:
 21:
 22:
 23:
 24:
 25:

P1 did not reach consensus
P2 has reached consensus (proposed 0, accepted 0)
